packages/@react-stately/button/src/ButtonGroupCollection.ts

```diff
@@ -17,9 +17,13 @@ export class ButtonGroupCollection<T extends ReactElement> implements Collection
   public items: T[];
   private keys: Key[];
 
-  constructor(items: T[]) {
+  constructor(items: T[], flipKeyOrder: boolean) {
     this.items = items;
     this.keys = items.filter(item => !item.props.isDisabled).map(child => child.key);
+    
+    if (flipKeyOrder) {
+      this.keys.reverse();
+    }
```

Comment:
Haven't looked into this but it feels like RTL logic handling this will need to happen in TreeCollection as well, but it seems to be a trickier case since each key stores what its next and last key is so a simple reverse doesn't work? Will look into later, prob in a different ticket